### PR TITLE
doc: update up_squared document to use UEFI bootable method.

### PR DESCRIPTION
**Pull request by chen-png**
for up_squared board, we already support to use UEFI bootable
method to run zephyr tests, so update the document to use
this UEFI method, rather than legacy BIOS stuff.
Now we have a UEFI loader, need to get rid of all the legacy BIOS stuff completely

Signed-off-by: Chen Peng1 <peng1.chen@intel.com>

### CODE REVIEW

**Comment by chen-png** [comment]
tested this UEFI method on up_squared in daily test system, it was good.

File: boards/x86/up_squared/doc/index.rst [hunk 1]
```diff
@@ -240,49 +181,25 @@ Prepare Linux host
...
 
       $ tree /srv/tftp
       /srv/tftp
-      ├── EFI
-      │   └── BOOT
-      │       └── grub.cfg
-      ├── grub_x86_64.efi
-      └── kernel
-          └── zephyr.strip
+      ├── zephyr.efi
```
Comment (finikorg) [inline boards/x86/up_squared/doc/index.rst:202]: If we want to have a directory tree we need to use "└──"

Comment (chen-png) [reply boards/x86/up_squared/doc/index.rst]: thanks, I have updated it.

**Review by finikorg — APPROVED** [review]
LGTM

**Comment by maksimmasalski** [comment]
Peng, rerun CI please.

**Comment by chen-png** [comment]
> Peng, rerun CI please.

yes, CI is ok now.

**Review by maksimmasalski — APPROVED** [review]

**Pull request merged**